modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/config/LifecycleAwareConfigurationInstance.java

```diff
@@ -158,6 +177,7 @@ public void dispose()
     {
         disposeIfNeeded(value, LOGGER);
         disposeIfNeeded(connectionProvider, LOGGER);
+        muleContext.fireNotification(new MetadataNotification(DISPOSE_METADATA_CACHE));
```

Comment:
on second thoughts, I think it's better to do this on the stop() rather than the dispose. Otherwise there could be some race conditions